src/common/IsolatedWorld.ts

```diff
@@ -170,11 +170,18 @@ export class IsolatedWorld {
 
   setContext(context: ExecutionContext): void {
     assert(
-      this.#contextPromise,
+      this.hasContext(),
       `ExecutionContext ${context._contextId} has already been set.`
     );
     this.#ctxBindings.clear();
-    this.#contextPromise.resolve(context);
+    this.#contextPromise.resolve(
+      (async () => {
+        if (this.#injected) {
+          await context.evaluate(injectedSource);
```

Comment:
I think it would work nicely if we don't await for context.evaluate. The CDP call will be scheduled and it will guarantee that scripts are injected before any other command reaches the renderer.